File: src/diffbot_py_pkg/diffbot_py_pkg/simple_publisher_example.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisherNode(Node):
    def __init__(self):
        super().__init__("simple_publisher")
        self.frequency_ = 1.0
        self.counter_ = 0.0
        self.simple_publisher = self.create_publisher(String, "chatter", 10)
        self.timer_ = self.create_timer(self.frequency_, self.publishCallback)

    def publishCallback(self):
        msg = String()
        msg.data = "Hi Ros2 : counter " + str(self.counter_)
        self.counter_ += 1.0
        self.simple_publisher.publish(msg)

def main():
    rclpy.init()
    simple_publisher_node = SimplePublisherNode()
    rclpy.spin(simple_publisher_node)
    simple_publisher_node.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/diffbot_cpp_pkg/src/simple_publisher_example.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

using namespace std::chrono_literals;

class SimplePublisherNode : public rclcpp::Node
{
public:
    SimplePublisherNode() : Node("simple_publisher")
    {
        simple_publisher_ = create_publisher<std_msgs::msg::String>("chatter", 10);
        timer_ = create_wall_timer(1s, std::bind(&SimplePublisherNode::publishCallback, this));
    }
private:

    void publishCallback()
    {
        std_msgs::msg::String msg;
        msg.data = "Hi Ros2 : counter " + std::to_string(counter_++);
        simple_publisher_->publish(msg);
    }

    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr simple_publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    unsigned int counter_;

};


int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    auto simple_publisher_node = std::make_shared<SimplePublisherNode>();
    rclcpp::spin(simple_publisher_node);
    rclcpp::shutdown();
    return 0;
}